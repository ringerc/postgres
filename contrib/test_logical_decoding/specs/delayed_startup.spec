setup
{
    CREATE EXTENSION test_logical_decoding;
    CREATE TABLE do_write(id serial primary key);
}

teardown
{
    DROP TABLE do_write;
    DROP EXTENSION test_logical_decoding;
    SELECT 'stop' FROM stop_logical_replication('isolation_slot');
}

session "s1"
setup { SET synchronous_commit=on; }
step "s1b" { BEGIN ISOLATION LEVEL SERIALIZABLE; }
step "s1w" { INSERT INTO do_write DEFAULT VALUES; }
step "s1c" { COMMIT; }
session "s2"
setup { SET synchronous_commit=on; }
step "s2init" {SELECT 'init' FROM init_logical_replication('isolation_slot', 'test_decoding');}
step "s2start" {SELECT data FROM start_logical_replication('isolation_slot', 'now', 'include-xids', 'false');}


permutation "s1b" "s1w" "s2init" "s1c" "s2start" "s1b" "s1w" "s1c" "s2start" "s1b" "s1w" "s2start" "s1c" "s2start"
