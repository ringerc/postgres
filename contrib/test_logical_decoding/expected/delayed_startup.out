Parsed test spec with 2 sessions

starting permutation: s1b s1w s2init s1c s2start s1b s1w s1c s2start s1b s1w s2start s1c s2start
step s1b: BEGIN ISOLATION LEVEL SERIALIZABLE;
step s1w: INSERT INTO do_write DEFAULT VALUES;
step s2init: SELECT 'init' FROM init_logical_replication('isolation_slot', 'test_decoding'); <waiting ...>
step s1c: COMMIT;
step s2init: <... completed>
?column?       

init           
step s2start: SELECT data FROM start_logical_replication('isolation_slot', 'now', 'include-xids', 'false');
data           

step s1b: BEGIN ISOLATION LEVEL SERIALIZABLE;
step s1w: INSERT INTO do_write DEFAULT VALUES;
step s1c: COMMIT;
step s2start: SELECT data FROM start_logical_replication('isolation_slot', 'now', 'include-xids', 'false');
data           

BEGIN          
table "do_write": INSERT: id[int4]:2
COMMIT         
step s1b: BEGIN ISOLATION LEVEL SERIALIZABLE;
step s1w: INSERT INTO do_write DEFAULT VALUES;
step s2start: SELECT data FROM start_logical_replication('isolation_slot', 'now', 'include-xids', 'false');
data           

step s1c: COMMIT;
step s2start: SELECT data FROM start_logical_replication('isolation_slot', 'now', 'include-xids', 'false');
data           

BEGIN          
table "do_write": INSERT: id[int4]:3
COMMIT         
?column?       

stop           
